--- helix/src/common/log/logsystem/hxtwritermanager.cpp Wed Dec 05 16:59:59 2007
+++ helix/src/common/log/logsystem/hxtwritermanager.cpp Wed Dec 05 16:59:59 2007
@@ -204,7 +204,7 @@
 		if( cPercent )
 		{
             const char* pszMsg = szMsg;
-#if defined (_SYMBIAN)
+//#if defined (_SYMBIAN)
             // fixup INT64 args (not supported by SYMBIAN vsprintf)
             CHXString str;
             if (strstr(szMsg, "%I64d"))
@@ -213,7 +213,7 @@
                 str.FindAndReplace("%I64d", HX_I64d_SUBST, TRUE);
                 pszMsg = str;
             }
-#endif
+//#endif
             CHXString strFormattedMsg;
             strFormattedMsg.FormatV(pszMsg, args);
             pMsgData->Message((const char*)strFormattedMsg);

--- helix/src/common/log/logobserverfile/hxtlogobserver.cpp Wed Dec 05 16:59:59 2007
+++ helix/src/common/log/logobserverfile/hxtlogobserver.cpp Wed Dec 05 16:59:59 2007
@@ -1128,7 +1128,7 @@
         {
             // DPRINTF_INIT(pPrefs);
             // Read the enabled pref
-            HXBOOL bEnabled = TRUE;
+            HXBOOL bEnabled = FALSE;    // Disabled by default
             ReadPrefBOOL(pPrefs, "Logging\\File\\Enabled", bEnabled);
 
             // Are we using queue/delivery thread mode?
